lib/compiler-llvm/src/abi/aarch64_systemv.rs

```diff
@@ -47,19 +49,36 @@ impl Abi for Aarch64SystemV {
         let param_types =
             std::iter::once(Ok(intrinsics.ctx_ptr_ty.as_basic_type_enum())).chain(user_param_types);
 
+        let vmctx_attributes = vec![
+            context.create_enum_attribute(Attribute::get_named_enum_kind_id("nofree"), 0),
+            context.create_enum_attribute(Attribute::get_named_enum_kind_id("nonnull"), 0),
+            context.create_enum_attribute(
+                Attribute::get_named_enum_kind_id("align"),
+                std::mem::align_of::<wasmer_vm::VMContext>()
+                    .try_into()
+                    .unwrap(),
+            ),
+        ];
+
         Ok(match sig.results() {
             [] => (
                 intrinsics
                     .void_ty
                     .fn_type(&param_types.collect::<Result<Vec<_>, _>>()?, false),
-                vec![],
+                vmctx_attributes
+                    .into_iter()
+                    .map(|x| (x, AttributeLoc::Param(0)))
+                    .collect(),
```

Comment:
This code seems repeated across the different match branches.
Why don't we move it up so we only write it once?

Reply:
Done.